lib/puma/launcher.rb

```diff
@@ -414,16 +414,14 @@ def setup_signals
         log "*** SIGINT not implemented, signal based gracefully stopping unavailable!"
       end
 
-      begin
-        Signal.trap "SIGHUP" do
-          if @runner.redirected_io?
+      if @runner.redirected_io?
+        begin
+          Signal.trap "SIGHUP" do
             @runner.redirect_io
-          else
```

Comment:
@grosser What I meant was you eliminated this branch of code, and I don't understand why.

Reply:
I was basically thinking "if you send SIGHUP to a server that does not have any reload mechanic then wtf are you doing" ... but maybe it's valid usecase :(

Reply:
I think it was important to #1377?